### PR TITLE
Refactor Spark DF read and Write options

### PR DESCRIPTION
**Pull request by karuppayya**
Refactor Spark DF read and Write options

### CODE REVIEW

File: spark/src/main/java/org/apache/iceberg/spark/SparkWriteOptions.java
Comment (aokolnychyi) [inline spark/src/main/java/org/apache/iceberg/spark/SparkWriteOptions.java:34]: I feel like there are more write properties. Is there a reason we don't include others?

File: spark/src/main/java/org/apache/iceberg/spark/SparkReadOptions.java
Comment (aokolnychyi) [inline spark/src/main/java/org/apache/iceberg/spark/SparkReadOptions.java:31]: There are also more read options. Why not include all of them?

File: spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java
Comment (aokolnychyi) [inline spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java:110]: Shall we use `SparkReadOptions.SNAPSHOT_ID` here?

**Comment by holdenk** [comment]
I like the idea of this change, would it make sense to update the documentation as well so folks see how to use it? Less prone to typos in the data read/write config path is <3 <3

**Comment by karuppayya** [comment]
@holdenk Thanks for feedback. I can add a column to site/docs/configuration.md's spark configs. Would that work?

**Comment by RussellSpitzer** [comment]
> @holdenk Thanks for feedback. I can add a column to site/docs/configuration.md's spark configs. Would that work?

I think you may want to add these to the Spark Section directly since they don't apply to configuration generally

**Comment by karuppayya** [comment]
> > @holdenk Thanks for feedback. I can add a column to site/docs/configuration.md's spark configs. Would that work?
> 
> I think you may want to add these to the Spark Section directly since they don't apply to configuration generally

I have added it as a "Note", open for any other suggestions for introducing into the doc.

**Review by holdenk — COMMENTED** [review]
LGTM, we'll need to wait for a committer to sign off though :)

File: site/docs/spark.md
Comment (rdblue) [inline site/docs/spark.md:364]: Can you remove "ex:" here and add the a language to the block so that it is highlighted correctly?

File: site/docs/spark.md
Comment (rdblue) [inline site/docs/spark.md:360]: I'm not sure that we want to encourage users to refer to constants in Iceberg. What do you think, @aokolnychyi, @holdenk, @RussellSpitzer?

File: site/docs/spark.md
Comment (rdblue) [inline site/docs/spark.md:551]: Same comment here as above.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Looks fine to me. Just a question for other people on the docs and a couple of nits in the docs.

File: site/docs/spark.md
Comment (karuppayya) [reply site/docs/spark.md]: done

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @karuppayya! Because there is still open discussion on whether we will recommend using the Iceberg constants, I've removed those docs from this PR and committed the rest.

**Pull request merged**
